AzureData/Source/AzureDataExtensions.swift

```diff
@@ -321,7 +321,14 @@ public extension User {
     }
 }
 
+// MARK: -
+
+public extension Permission {
```

Comment:
Actually, can we just put a generic `refresh` on the `CodableResource` extension and remove all `refresh`s from the resource-type specific extensions?  i.e.

```swift
public extension CodableResource {
    
    public func delete(_ callback: @escaping (Response<Data>) -> ()) {
        return DocumentClient.shared.delete(self, callback: callback)
    }

    public func refresh(_ callback: @escaping (Response<Self>) -> ()) {
        return DocumentClient.shared.refresh(self, callback: callback)
    }
    
    public func create(permissionWithId permissionId: String, mode permissionMode: PermissionMode, forUser user: User, callback: @escaping (Response<Permission>) -> ()) {
        return DocumentClient.shared.create(permissionWithId: permissionId, mode: permissionMode, in: self, forUser: user, callback: callback)
    }
}
```

Reply:
Yup. Smart!